server/models/schema.sql

```diff
@@ -1873,11 +1873,11 @@ CREATE TABLE IF NOT EXISTS `voucher` (
   `uuid` char(36) NOT NULL,
   `date` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
   `project_id` tinyint(4) NOT NULL,
-  `reference` varchar(16) NOT NULL,
+  `reference` varchar(16) DEFAULT NULL,
```

Comment:
Interesting.  This should be an automatically generated integer.  Could you change this to:
```sql
  `reference`       INT(10) UNSIGNED NOT NULL DEFAULT 0,
```
and add the following TRIGGER
```sql
CREATE TRIGGER voucher_calculate_reference BEFORE INSERT ON voucher
FOR EACH ROW SET NEW.reference = (SELECT IFNULL(MAX(reference) + 1, 1) FROM voucher WHERE voucher.project_id = NEW.project_id);
```
It apparently got through the first review without correction.